src/components/views/rooms/NotificationBadge/StatelessNotificationBadge.tsx

```diff
@@ -56,17 +59,20 @@ export const StatelessNotificationBadge = forwardRef<HTMLDivElement, XOR<Props,
             symbol = formatCount(count);
         }
 
+        // We show a dot if either:
+        // * The props force us to, or
+        // * It's just an activity-level notification or (in theory) lower and the room isn't knocked
+        const badgeType = forceDot || (level <= NotificationLevel.Activity && !knocked) ? "dot" : "badge";
```

Comment:
can we extend this to 2 vs 3 char badges, so we don't have to repeat the `symbol` logic?

```suggestion
        const badgeType = forceDot || (level <= NotificationLevel.Activity && !knocked) ? "dot" : 
            (!symbol || symbol.length < 3) ? "badge_2char" : "badge_3char"
```

Reply:
Incorporated separately